ci/lava/tests/test-config-across-reboot.py

```diff
@@ -0,0 +1,190 @@
+
+    def test_change_hostname(self, execute_helper):
+        """Change the hostname.
+
+        Append some random digits to the current hostname.
+        """
+        TestConfigAcrossReboot.newHostname = "{}-{}".format(
+            TestConfigAcrossReboot.originalHostname, self._randomStringDigits()
+        )
+        print(TestConfigAcrossReboot.newHostname)
+
+        """Perform the test on the DUT via the mbl-cli."""
+        err, stdout, stderr = execute_helper.send_mbl_cli_command(
+            [
+                "shell",
+                'sh -l -c "echo {} > /config/user/hostname"'.format(
+                    TestConfigAcrossReboot.newHostname
+                ),
+            ],
+            TestConfigAcrossReboot.dut_address,
+        )
+        assert err == 0
+
+    def test_get_wlan_ip(self, execute_helper):
+        """Perform the test on the DUT via the mbl-cli."""
+        err = self._disable_technology("ethernet", execute_helper)
+        assert err == 0
+
+        err, TestConfigAcrossReboot.wlan_ip_address = self._get_wlan_ip(
+            execute_helper
+        )
+        self._ifconfig(execute_helper)
+        assert err == 0
+        assert TestConfigAcrossReboot.wlan_ip_address is not None
+
+    def test_reboot_dut(self, dut_addr, execute_helper):
+        """Cause the DUT to reset."""
+        err, stdout, stderr = execute_helper.send_mbl_cli_command(
+            ["shell", 'sh -l -c "/sbin/reboot"'],
+            TestConfigAcrossReboot.dut_address,
+        )
+        # The err code varies and is either 0 of 255 so we cannot depend on it.
+        # Making the test pass is reasonable since if the reboot doesn't happen
+        # the hostname will not have changed and the next step will fail.
+        pass
+
+    def test_dut_online_after_reboot(self, dut, execute_helper):
+        """Wait for the DUT to come back online.
+
+        The previous test cause the DUT to reboot.
+        """
+        dut_address = ""
+
+        # Wait some time before discovering again the DUT
+        time.sleep(20)
+
+        while not dut_address:
+            dut_address = get_dut_address(dut, execute_helper)
+            if dut_address:
+                print("DUT {} is online again".format(dut_address))
+                break
+            print("DUT is still offline. Trying again...")
+        assert dut_address is not None
+
+    def test_get_new_hostname(self, dut, dut_addr, execute_helper):
+        """Get the hostname from the DUT.
+
+        Check the new hostname matches the new version.
+        """
+        err, hostname = self._get_hostname(dut, dut_addr, execute_helper)
+
+        assert err == 0
+        assert hostname is not None
+        assert TestConfigAcrossReboot.newHostname is not None
+        assert hostname == TestConfigAcrossReboot.newHostname
+
+    def test_get_wlan_ip_after_reboot(self, execute_helper):
+        """Perform the test on the DUT via the mbl-cli."""
+        self._ifconfig(execute_helper)
```

Comment:
Why this?

Reply:
This is debug code that will show the ifconfig of the DUT in the Lava Log - might be useful to understand the reason if the IP address part of the test fails.